Updated docs for roles and permissions

noosheenv opened the pull request:
## Internal Notes for Reviewers
I updated the docs for Roles and Permissions to align with the UI:
- Split `manage-roles-and-permissions.qmd` into `manage-roles.qmd` and `manage-permissions.qmd`
- Updated the sources to match the names: `_manage-roles.qmd` and `_manage-permissions.qmd` from the really long name before
- Removed the "Add roles" section as the feature will not be implemented for the next 3 sprints.
- Tested the old steps and modified them to match the current UI
- Changed the prerequisites from `Superuser or Group Admin` to `Customer Admin` for accuracy

I would love some opinions and advice, especially for:
- Differentiating intros for the pages (currently just copied from the old doc)
- Since splitting means file name changes, references to the old doc in the Customer Admin training were updated. 
  - Please check http://localhost:7503/training/training-for-administrators.html#/section-6 (or similar) if this still makes sense
- The last step for Manage roles, "6. Select **Users** to manage users assigned to the role."
  - I've been undecided on the wording, between "users with the role", "users under the role", and others. 
<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `deprecation`
- `documentation`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## External Release Notes

<!--- REPLACE THIS COMMENT WITH YOUR DESCRIPTION --->

validbeck commented on site/guide/manage-roles.qmd:
```diff
@@ -1,32 +1,32 @@
 1. [Log in to the {{< var vm_platform >}}]({{< var vm_url >}}).
 
-2. Click **Settings**.
+1. Click **Settings**.
```
```suggestion
1. Click **{{< fa gear >}} Settings**.
```

validbeck commented on site/guide/manage-permissions.qmd:
```diff
@@ -0,0 +1,47 @@
+
+1. [Log in to the {{< var vm_platform >}}]({{< var vm_url >}}).
+
+1. Click **Settings**.
```
```suggestion
1. Click **{{< fa gear >}} Settings**.
```

validbeck commented on site/guide/_manage-roles.qmd:
```diff
@@ -0,0 +1,21 @@
+
+1. Select the role for which you want to manage.
+
+1. Select **Edit Permissions** to edit its details. Within the role's details, you can:
```
In general, we want to stay away from the nebulous "it" – simply state what "it" is. 

e.g. "Click **Edit Permissions** to add or remove permissions to or from that role." 

I would try and break this section down into the "assign permissions to roles" and "add users to roles" as these are actually separate tasks. You'll need to make sure it works with the training as well.

validbeck commented on site/guide/_manage-roles.qmd:
```diff
@@ -0,0 +1,21 @@
+1. Under User Permissions, select **Roles**.
```
I would move the list you have under 5. here. You're also missing some information about how you can see the list of users currently assigned to this role which is actually the core part of the article, since this is the one about roles and not permissions! (Even though roles HAVE permissions...)

validbeck commented on site/guide/_manage-roles.qmd:
```diff
@@ -0,0 +1,21 @@
+   - Remove permissions that are no longer necessary for the role.
+   - Save the updated permissions configuration for the role.
+
+1. Select **Users** to manage users assigned to the role.
```
Pretty sure we'd like some step-by-step here, as this is the managing roles article. ;)

validbeck commented on site/guide/_manage-permissions.qmd:
```diff
@@ -0,0 +1,3 @@
+1. Under User Permissions, select **Permissions**
+
+1. Under your desired action, add or delete roles. 
```
How do I this fam, I've never seen the interface before, where do I click, give me some more context, please give me a step-by-step, or maybe even just break it down into bullet points:

e.g.
```
Manage roles that have authorization to perform actions:

- To assign a role to a permission, blah blah blah
- To remove a role from a permission, blah blah blah```

nrichers replied:
To add to what Beck said, the steps should cover that you click **+** to add a role to the permission or remove a role by clicking on the **x** on the label: 

<img width="991" alt="image" src="https://github.com/validmind/documentation/assets/15148011/b42c1883-8afa-498e-b86e-7add2252b60b">

validbeck reviewed (changes requested):
> A general comment about PRs, I would recommend providing screencaps for "before" or "after" especially for sections of the docs site – not everyone knows immediately where to look for the changes if they're not clearly indicated, and sometimes PRs may need to be reviewed by folks not on the docs team.

# Global comments
- needs moar context
- we want some step-by-step

## Onboarding users
> Since `Onboarding users` is technically a landing page with subpages, I would put a "What's next" section down at the bottom with the title cards from this section! You might need to edit the summaries so they look good on the CTA. ;) 

## Manage roles
> There's another catch (edge-case) here I think is worth noting on this page for users – take a look at the UI again and see if you can figure out what it is.

## Training for administrators
> Please check http://localhost:7503/training/training-for-administrators.html#/section-6 (or similar) if this still makes sense

That's YOUR local URL, won't work for anyone else. I would recommend linking to the line in the code of your branch. e.g. "Refer to [site/training/training-for-administrators.qmd](https://github.com/validmind/documentation/blob/4486f0f9d0faf5633d59e6d6fe12a9e372bd3605/site/training/training-for-administrators.qmd#L183) 

## Role name cleanup
> Changed the prerequisites from Superuser or Group Admin to Customer Admin for accuracy

I would also make sure all OTHER instances of referring to a superuser or group admin are removed on the other docs.

nrichers commented on site/guide/_manage-permissions.qmd:
```diff
@@ -0,0 +1,3 @@
+1. Under User Permissions, select **Permissions**
```
```suggestion
1. Under User Permissions, select **Permissions**.
```

nrichers commented on site/guide/_manage-permissions.qmd:
```diff
@@ -0,0 +1,3 @@
+1. Under User Permissions, select **Permissions**
+
+1. Under your desired action, add or delete roles. 
```
```suggestion
1. Add or delete roles that hold a specific permission:
```

I would then include the bulleted list you have in your comment/question and add an example below that: 

```
   - To assign a role to a permission, blah blah blah
   - To remove a role from a permission, blah blah blah

   For example, to grant a developer the permission to update an existing template ... yada yada yada 
```

Also, wording like `Under your desired action` doesn't really add anything, I'd just skip in favour of direct wording — what do I desire if I am a new user?

nrichers commented on site/guide/_manage-roles.qmd:
```diff
@@ -0,0 +1,21 @@
+1. Under User Permissions, select **Roles**.
+
+1. Select the role for which you want to manage.
```
This sentence needs a visit from your friendly neighbourhood grammar fairy. 🧚

nrichers commented on site/guide/manage-permissions.qmd:
```diff
@@ -0,0 +1,47 @@
+[Summarize the task and provide any next steps or resources for the user to continue their learning or work.]
+--->
+
+## What's next
```
As a comment on both "Manage roles" and "Manage permissions", I'm not sure these related links add value — I would just remove. We already get some of this for free via the bottom links (see the screenshot) and these seem like parallel rather than sequential tasks. 

<img width="1037" alt="image" src="https://github.com/validmind/documentation/assets/15148011/3932b6ff-7f12-4724-b64a-7c3bcc805016">

nrichers reviewed (commented):
Another great PR from you, @noosheenv! Don't let the comments from Beck and I distract from this — our comments are made through the lens of people who look at instructions every day. (Also, I'm never quite sure if the "we" in Beck's comments is the royal form or just a generic reference. 😛 )

> Differentiating intros for the pages (currently just copied from the old doc)

My recommendation would be to rewrite both of these without the added `These steps guide you through the process of ...` as that's just verbiage you have to wade through. Be direct and tell the reader they are doing (or, as Beck would say, write in the imperative.)

For context, I've gradually been removing similar extra verbiage in other topics as it's unnecessary. For topics that get included as content tiles on listing pages, it generally also just looks off because it's so repetitive. Similarly, the extra verbiage affects what you see in search results.

> Since splitting means file name changes, references to the old doc in the Customer Admin training were updated.

I checked and the training looks correct, very nice.

validbeck reviewed (changes requested):
@noosheenv Much better! In the interest of time I just made the following edits for you:

- Important callouts should go BEFORE the thing they're warning about changing, generally. 
- You still need to break instructions down more granularly. See the changes I made in "Manage roles" as to what I'm talking about. Anything that is a SEPARATE task should have it's own (sub-)section, and preliminary directions, unless it's something small like a add/edit menu that looks exactly the same, just opened with a slightly different button. 
- When you're writing and editing instructions, make sure you're reviewing them for correct emphasis and so on as per the style guide (see the changes in "Manage permissions"). 

| Manage roles | Manage permissions |
|---|---|
| <img width="790" alt="Screenshot 2024-06-20 at 11 34 11 AM" src="https://github.com/validmind/documentation/assets/164545837/9c68308a-0a35-4c73-b61e-f5dd04820e8e">| <img width="1008" alt="Screenshot 2024-06-20 at 11 33 55 AM" src="https://github.com/validmind/documentation/assets/164545837/363f6d91-ed96-4339-b56f-e5020ad031b2"> |

> We can leave the changes to the group admin/superuser for another Story (please make that Story now if you haven't, and link it to this existing Story via the related Stories), but I'd like you to quickly add the `What's next` section at the bottom of the "Onboarding users" landing page before calling this one completed. You may need to edit the CTAs so that the cards look pretty.

noosheenv commented:
Thanks for making those changes! I made the new story and linked it to this one.

I added the What's next, cards, and deleted the example section (even though I should've saved it for the other story, I just couldn't stand it LMAO.)

I edited the CTAs to look like the other landing pages, not sure if that means it's pretty enough though.

validbeck reviewed (changes requested):
@noosheenv Oops, sorry, I just realised my push alias is a force push, I didn't mean to write over you pulling in things from `main`! MB there.

- Thank you for removing the example, and adding the "What's next" section!
- The CTAs/summaries still need a lot of love... for example, "Manage users" trails off and is the clunkiest most roundabout lil thing ever! Make sure these descriptions are under the character limit, and let's just take out the "This task topic will ..." as this is a needlessly roundabout way to talk about something. Just state what's involved without the fluff. Summarising is also GREAT writing practice. ;) Following the rules for active voice and direct language, I'd include (not necessarily in the same order every time, play around with sentence structure):
   - In `x` context
   - Do `y` task
   - Because `z` value
 - There's also some accuracy issues now that we've moved some of the information under this section around, so pay attention to what you're actually summarizing. 

e.g. 
```
Invite new users and assign users to specific groups and roles to maintain access control and efficient user management within your organization via the {{< var vm_platform >}}.
```

noosheenv commented:
<img width="985" alt="image" src="https://github.com/validmind/documentation/assets/143854391/f5410727-2278-4519-8b74-9980177a913c">
Thanks for the template, it really helped! I used chatGPT by pasting in the contents of the guide, then asking it to make an intro for that feature, which helps for accuracy. Then, I edited it to follow the guidelines given above. 

From here, further changes to other guides' CTA's may be out of the scope of this roles/permissions PR, and it may be best to tack it onto the other story we created.

validbeck commented:
@noosheenv I just pushed a quick edit for the summaries, but otherwise it does look good to go (no further changes required from your end)!
